src/perftest_communication.c

```diff
@@ -1062,25 +1064,24 @@ int rdma_server_connect(struct pingpong_context *ctx,
 	struct rdma_conn_param conn_param;
 	struct addrinfo hints;
 	char *service;
-	struct sockaddr_in sin;
+	struct sockaddr sin;
```

Comment:
here is a bug, you can't define a `struct sockaddr` variable to save the ipv6 address. `rdma_bind_addr()` fails with the current change.
change it to `struct sockaddr *sin`, and point it to `res->ai_addr`, no need to copy the values.
also it looks line 1083 (sockaddr_set_port) is not necessary as `check_add_port` already assigns the port.

Reply:
`rdma_client_connect()` has the same bug.